src/oidc/oidc.util.ts

```diff
@@ -0,0 +1,190 @@
+import { Configuration } from 'oidc-provider';
+
+let Provider, oidc;
+const configuration: Configuration = {
+  // ... see the available options in Configuration options section
+  features: {
+    // claimsParameter:{
+    //     enabled: false
+    // },
+
+    /* enables grant_type=client_credentials to be used on token endpoint */
+    clientCredentials: {
+      enabled: true,
+    },
+    resourceIndicators:{
+      enabled: true,
+      getResourceServerInfo(ctx,resourceIndicator,client){
+        if(resourceIndicator)
+          return {
+            scope: 'openid email profile phone offline_access address',
+            audience: resourceIndicator,
+            accessTokenTTL: 1*60*60,
+            accessTokenFormat: 'jwt'
+          }
+        throw new Error("LOL")
+      }
+    },
+    /* To extract out the info in the tokens generated, default has to be modified to allow introspection from authentic sources only (features.introspection.allowedPolicy)*/
+    introspection: {
+      enabled: true,
+      // this is default function, change it
+      // allowedPolicy(ctx, client, token) {
+      //   if (client.clientAuthMethod === 'none' && token.clientId !== ctx.oidc.client.clientId) {
+      //     return false;
+      //   }
+      //   return true;
+      // },
+    },
+    /* resourceIndicators: {
+       enabled: true,
+        getResourceServerInfo(ctx, resourceIndicator) {
+        if (resourceIndicator === 'urn:api') {
+           return {
+             scope: 'read',
+             audience: 'urn:api',
+             accessTokenTTL: 1 * 60 * 60, // 1 hour
+             accessTokenFormat: 'jwt',
+           };
+         }
+
+         throw new Error('Invalid target');
+       },
+     }, */
+    /* To turn off default pages and routes */
+    devInteractions: {
+      enabled: false,
+    },
+  },
+  interactions: {
+    url(_, interaction) {
+      return `/interaction/${interaction.uid}`;
+    },
+  },
+  clientBasedCORS(ctx, origin, client) {
+    return false;
+  },
+
+  //   claims:{
+  //     profile: ['username','gender','birthdate','email'],
+  //   },
```

Comment:
remove unless this is for document purposes, if this is for document purposes, change this to use JSDoc format.